### PR TITLE
Use webnative-elm

### PR DESCRIPTION
**Pull request by matheus23**
The bigger picture plan is addressing #101 using webnative-elm (which was built as a result of that issue).

For this codebase concretely this means:
* Building a secondary `Msg` structure, we call `Tag` (subject to renaming!)
* Adding the `FileSystem.Actions` module for automating encoding/decoding the tags into/from strings
* Removing the need for `FileSystem.State`.

### CODE REVIEW

**Comment by matheus23** [comment]
So far, I've built the plumbing, and used it in one place (the save button for the plaintext editor).
Rest is TODO

File: elm-analyse.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "checks": {
+        "UnusedPatternVariable": false
+    },
+    "exclude": [ "src/Library/Tailwind.elm" ]
```
Comment (matheus23) [inline elm-analyse.json:5]: This makes the ElmLS more responsive for me. Figured, I could leave this in here for others to enjoy until ElmLS performance issues are resolved.

File: src/Javascript/index.js [hunk 2]
```diff
@@ -11,23 +11,18 @@
...
+import"./web_modules/webnative-elm.js"
 
 import "./custom.js"
 
 import * as analytics from "./analytics.js"
 import * as fs from "./fs.js"
 import * as ipfs from "./ipfs.js"
-import * as media from "./media.js"
-import { throttle } from "./common.js"
-import { setup } from "./sdk.js"
```
Comment (matheus23) [inline src/Javascript/index.js:22]: These imports were unused, so I removed them

File: src/Javascript/index.js [hunk 3]
```diff
@@ -11,23 +11,18 @@
...
 
 
 // | (• ◡•)| (❍ᴥ❍ʋ)
 
 
-self.wn = wn
-
-
```
Comment (matheus23) [inline src/Javascript/index.js:30]: Not sure whether I should have actually removed this. What's its use?

**Comment by matheus23** [comment]
@icidasset I haven't replaced all the calls, yet. For now the only things which were replaced with webnative-elm calls were sidebar stuff.
So
* Creating empty files/folders
* Saving/loading text files for the plaintext editor

Peeking at the `listDirectory` javascript code, I think that function will stay for a little longer, as it includes some file size logic, too.

All in all I think this is in a review-ready stage, even when it doesn't 100% replace everything with webnative-elm yet. But I'd rather talk through changes early before I do a bigger amount of work in the wrong direction :smile: 

Also, this already improves some thing greatly! It fixes a bug with the save button and enables implementing *actual* loading indicators for creating empty files/folders.

File: src/Javascript/index.js (hunk 3 above)
Comment (icidasset) [reply src/Javascript/index.js:30]: Just to make it available in the browser console.
(`self = window` in the ui context)

File: elm-analyse.json (hunk 1 above)
Comment (icidasset) [reply elm-analyse.json:5]: Yeah that's fine, shouldn't be analysed anyway.

**Review by icidasset — APPROVED** [review]

**Pull request merged**